sklearn/manifold/_spectral_embedding.py

```diff
@@ -189,7 +189,9 @@ def spectral_embedding(
         of the lobpcg eigen vectors decomposition when `eigen_solver ==
         'amg'`, and for the K-Means initialization. Use an int to make
         the results deterministic across calls (See
-        :term:`Glossary <random_state>`).
+        :term:`Glossary <random_state>`). If None, the random number
+        generator is the ``RandomState`` instance used by ``np.random``,
+        that is ``np.random.default_rng()``.
```

Comment:
### For information purposes

`np.random.default_rng()` returns a NumPy [Generator](https://numpy.org/doc/stable/reference/random/generator.html) which is not a `RandomState` object.

With `None`, we use the "global NumPy random state object", which does not have a public API. We directly access it here:

https://github.com/scikit-learn/scikit-learn/blob/432778464cbffc8ca675c1df786c31f8c23fc62c/sklearn/utils/validation.py#L1156-L1157